### PR TITLE
Fix directory names and add table of contents

### PR DESCRIPTION
**Pull request by Cito**
- make the names of the subdirectories more consistent: replaced underscores with dashes since that was what most of them were using
- add script to print a table of contents
- add table of contents to the readme

### CODE REVIEW

File: create_toc.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                title = line[1:].strip()
+                break
+        else:
+            title = None
+        if not title:
+            raise RuntimeError(f"No title found in {spec_path}")
+        try:
+            desc, name = title.rsplit("(", 1)
+            desc = desc.rstrip()
+            name = name[:-1].strip().capitalize()
```
Comment (TheByronHimes) [inline create_toc.py:33]:
```suggestion
            name = name[:-1].strip().title()
```
I think title case looks a little nicer with multi-word titles (Wood Ant vs Wood ant). It's a style choice so take it or leave it.

File: create_toc.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            num = int(epic_dir.split("-", 1)[0])
+            if not 0 <= num < 1000:
+                raise ValueError
+        except ValueError:
+            raise RuntimeError(f"Invalid directory name {epic_dir}")
+        spec_path = os.path.join(epic_dir, "technical_specification.md")
+        if not os.path.exists(spec_path):
+            raise RecursionError(f"No technical_specification.md found in {epic_dir}")
+        for line in open(spec_path):
+            if line.startswith("#"):
```
Comment (TheByronHimes) [inline create_toc.py:23]:
```suggestion
            if line.lstrip().startswith("#"):
```
If you're bothering to strip the name below, then you might as well strip any spaces that wind up in front of the line so it doesn't try to read the scope header as the title.

**Review by TheByronHimes — CHANGES_REQUESTED** [review]
One small change I think should be included and one suggestion. 
This seems like something that could be a pre-commit hook, too.

File: create_toc.py (hunk 2 above)
Comment (Cito) [reply create_toc.py]: Actually the line needs to start with a `#` but I get your point and fixed it in 6cbe3da.

**Comment by Cito** [comment]
If the line numbering is non-continuous, then Markdown makes it continuous which is not what we want. Made it an unordered list in that case to preserve the original numbering.

**Comment by Cito** [comment]
We can consider using a pre-commit hook and a script that auto-generates the readme after moving to the GitLab based documentation.

**Review by TheByronHimes — APPROVED** [review]

**Pull request merged**